src/render_tree/properties.rs

```diff
@@ -10,19 +10,20 @@ pub struct Rectangle {
 
 impl Default for Rectangle {
     fn default() -> Self {
+        Self::new()
+    }
```

Comment:
Couldn't you use `#[derive(Default)]` here because the default rectangle uses zeros for the new method?

Reply:
I guess I could, I just wanted to be consistent with the other default implementations. A pro imo is that, if you change the `new` constructor to accept parameters, then you are forced to reason about what the default implementation should do, or you even need one. If we derive the trait, then this connection is lost and this could lead to surprises in scenarios where you called `default` instead of `new`.